lib/middleman-search_engine_sitemap/extension.rb

```diff
@@ -49,14 +41,7 @@ def not_excluded?(resource)
       end
 
       def page_ext
-        File.extname(app.index_file)
-      end
-
-      def register_extension_templates
-        # We call reload_path to register the templates directory with Middleman.
-        # The path given to app.files must be relative to the Middleman site's root.
-        templates_dir_relative_from_root = Pathname(TEMPLATES_DIR).relative_path_from(Pathname(app.root))
-        app.files.reload_path(templates_dir_relative_from_root)
```

Comment:
This was my workaround for https://github.com/middleman/middleman/issues/1217. It's a hard thing to test for, and there is no test for it at the moment. The problem arises when this gem gets placed in the `vendor` directory of a Middleman site (platforms like Heroku do this).